src/coreComponents/dataRepository/Utilities.cpp

```diff
@@ -0,0 +1,183 @@
+
+
+#include "common/MpiWrapper.hpp"
+#include "Utilities.hpp"
+#include "Group.hpp"
+
+
+#include <unordered_set>
+#include <unordered_map>
+
+namespace geosx
+{
+namespace dataRepository
+{
+
+void printMemoryAllocation( Group const & group, integer const indent, real64 const threshold )
+{
+  // static flag to keep track of whether or not a subgroup is the last one, and thus
+  // visually will terminate the tree.
+  static bool terminateBranch[64]{};
+
+  // keep track of the address of the groups that have been printed. Not the best
+  // way to do this, but doens't require infrastructure changes.
+  static std::unordered_set< void const * > groupsPrinted;
+
+  // initialize the static variables at the head of the tree.
+  if( indent==0 )
+  {
+    terminateBranch[0] = true;
+    for( int i=1; i<64; ++i )
+    {
+      terminateBranch[i] = false;
+    }
+    groupsPrinted.clear();
+  }
+
+  // store the local allocations for each wrapper
+  std::vector< double > localAllocations;
```

Comment:
`double` seems an unorthodox choice for storing byte sizes. Would a 64-bit int (`size_t`?) work for all of these, or are we planning to allocate more than 18 EB in total across some large machines?

Reply:
In an earlier implementation, I was doing scalings on `localAllocations`. At that time, the variable was being output as a floating point number once it is scaled (I was originally using a fixed scaling to `MB`). I guess we can change it back, but it eventually gets run through a reduction with the average of the reduction being a floating point number. And then all allocation numbers would have to be converted to a floating point for printing??

I guess i still have a preference to convert it to a `double` at the beginning rather than at he end. Do you think I should change it to `size_t`? What would the danger be of the way it is done? Perhaps someone would try to use this number assuming it is `size_t`. Maybe I my thinking is backwards.